### DISCUSSION TITLE
Android or React Native compatibility

### DISCUSSION THREAD

**Discussion by ninjz** [donmccurdy/glTF-Transform#482] [General]
Is there any way to make this work on Android/React-Native apps? I'd like to have the transformations performed on the client side.

**Comment by donmccurdy** [comment]
I don't use React Native myself and can't give a categorical yes/no here, but I'll summarize the dependencies of the project's modules and let you decide. The answer probably depends on which transformations you need to access.

- **/core** is plain JavaScript and should run anywhere that JavaScript runs. The NodeIO, WebIO, and DenoIO utilities for reading/writing files rely on APIs of the respective platforms. If React Native doesn't provide any of those APIs, you might need a custom I/O implementation. If you can get the GLB as a Uint8Array client-side, then I think WebIO is likely to work just fine in React Native.
- **/extensions** is plain JavaScript, with two dependencies if you need to support [Draco](https://gltf-transform.donmccurdy.com/classes/extensions.dracomeshcompression.html) or [Meshopt](https://gltf-transform.donmccurdy.com/classes/extensions.meshoptcompression.html) compression. Both compression libraries are provided as WASM binaries by their authors, but you could probably compile them to ASM.js as well.
- **/functions** is a mix of plain JavaScript utilities and a few that have dependencies as described in the [Functions documentation](https://gltf-transform.donmccurdy.com/functions.html). Functions that alter image data depend on [ndarray-pixels](https://github.com/donmccurdy/ndarray-pixels/) which works in Node.js or Web, but I don't know about React Native. It may need a Canvas 2D implementation there.
- **/cli** has many dependencies on the Node.js environment and is unlikely to run elsewhere.

If the transformations you need to make are things like adjusting materials and changing vertex data, then /core and /extensions are probably all you need. If you need additional compression or transforms from the /functions package I think you'll need to inspect their dependencies individually.

**Reply by ninjz** [reply]
Thank you for your very detailed response. I will give your suggestion a try for getting the GLB as a Uint8Array. I found this lib here: [bwasty/gltf-loader-ts](https://github.com/bwasty/gltf-loader-ts) which looks helpful.

My main goal is to merge multiple GLB files which contain all the exact same nodes, meshes, and materials, but have a different animation. Basically forming one GLB with multiple animations at runtime instead of loading one GLB file (which could contain 150+ animations) every time.

I've gotten this to work locally using Node.js and only using /core and /functions-{prune, dedup} so this looks promising.

**Reply by Osama217** [reply]
Hi, 

Were you able to get gltf-transform working in react-native?
